cellrank/tl/estimators/mixins/decomposition/_eigen.py

```diff
@@ -0,0 +1,378 @@
+                "D": D,
+                "stationary_dist": pi,
+                "V_l": V_l,
+                "V_r": V_r,
+                "eigengap": e_gap,
+                "params": {"which": which, "k": k, "alpha": alpha},
+            },
+            time=start,
+        )
+
+    @d.dedent
+    def plot_spectrum(
+        self,
+        n: Optional[int] = None,
+        real_only: bool = False,
+        show_eigengap: bool = True,
+        show_all_xticks: bool = True,
+        legend_loc: Optional[str] = None,
+        title: Optional[str] = None,
+        marker: str = ".",
+        figsize: Optional[Tuple[float, float]] = (5, 5),
+        dpi: int = 100,
+        save: Optional[Union[str, Path]] = None,
+        **kwargs: Any,
+    ) -> None:
+        """
+        Plot the top eigenvalues in real or complex plane.
+
+        Parameters
+        ----------
+        n
+            Number of eigenvalues to show. If `None`, show all that have been computed.
+        real_only
+            Whether to plot only the real part of the spectrum.
+        show_eigengap
+            When ``real_only = True``, this determines whether to show the inferred eigengap as
+            a dotted line.
+        show_all_xticks
+            When ``real_only = True``, this determines whether to show the indices of all eigenvalues on the x-axis.
+        legend_loc
+            Location parameter for the legend.
+        title
+            Title of the figure.
+        marker
+            Marker symbol used, valid options can be found in :mod:`matplotlib.markers`.
+        %(plotting)s
+        kwargs
+            Keyword arguments for :func:`matplotlib.pyplot.scatter`.
+
+        Returns
+        -------
+        %(just_plots)s
+        """
+
+        eig = self.eigendecomposition
+        if eig is None:
+            raise RuntimeError(
+                "Compute eigendecomposition first as `.compute_eigendecomposition()`."
+            )
+
+        if n is None:
+            n = len(eig["D"])
+        if n <= 0:
+            raise ValueError(f"Expected `n` to be > 0, found `{n}`.")
+
+        if real_only:
+            fig = self._plot_real_spectrum(
+                n,
+                show_eigengap=show_eigengap,
+                show_all_xticks=show_all_xticks,
+                dpi=dpi,
+                figsize=figsize,
+                legend_loc=legend_loc,
+                title=title,
+                marker=marker,
+                **kwargs,
+            )
+        else:
+            fig = self._plot_complex_spectrum(
+                n,
+                dpi=dpi,
+                figsize=figsize,
+                legend_loc=legend_loc,
+                title=title,
+                marker=marker,
+                **kwargs,
+            )
+
+        if save:
+            save_fig(fig, save)
+
+    def _plot_complex_spectrum(
+        self,
+        n: int,
+        dpi: int = 100,
+        figsize: Optional[Tuple[float, float]] = (None, None),
```

Comment:
```suggestion
        figsize: Tuple[Optional[float], Optional[float]] = (None, None),
```

Reply:
Changed to `Optional[Tuple[float, float]]`, since that is the correct type.